gen.go

```diff
@@ -0,0 +1,215 @@
+package avro
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"go/ast"
+	"go/token"
+	"io"
+	"strings"
+
+	"github.com/iancoleman/strcase"
+	"golang.org/x/tools/go/ast/astutil"
+)
+
+type GenConf struct {
+	PackageName string
+	OutFileName string
+}
+
+var primitiveMappings = map[Type]string{
+	"string":  "string",
+	"bytes":   "[]byte",
+	"int":     "int",
+	"long":    "int64",
+	"float":   "float32",
+	"double":  "float64",
+	"boolean": "bool",
+}
+
+func GenerateFrom(gc GenConf, s string) (io.Reader, error) {
```

Comment:
Logically i would swap these?
```suggestion
func GenerateFrom(s string, gc GenConf) (io.Reader, error) {
```
Also seems odd that you can have an `OutFile` and return a reader. Perhaps just an `io.Writer` could be supplied and the user could pick what to do with it. The `cmd` would obviously supply a file.